src/flashbar/__integ__/collapsible.test.ts

```diff
@@ -78,7 +79,7 @@ describe('Collapsible Flashbar', () => {
       test(
         'adding flash with ariaRole="alert" moves focus to the new flash',
         setupTest(async page => {
-          await page.toggleCollapsibleFeature();
+          await page.toggleStackingFeature();
           await page.toggleCollapsedState();
           await wait(FOCUS_THROTTLE_DELAY);
```

Comment:
What happens if user clicks two buttons too quick?

What happens if our consumers do not add this timeout in their integration tests?

Does the stacking feature respect our disable-motion config? (which is activated in tests and makes all transitions instant)

Reply:
Note that the delay is for the focus behavior, not for items to be added/removed or animations. The focus is throttled with a delay in order to give time for screen readers to read an item before focusing on another one. This was implemented as part of [#454](https://github.com/cloudscape-design/components/pull/454). As explained in that PR's description:

> There is also throttling for focus moves, to prevent abrupt jumps when multiple messages are added sequentially

So to answer your questions:

> What happens if user clicks two buttons too quick?

The items will be added or removed as expected. The focus would be debounced so that screen reader messages do not overlap too much, as explained before.

> What happens if our consumers do not add this timeout in their integration tests?

If they use utility functions like `getItems`, they should return the expected results instantaneously as usual. If they are asserting focus behavior, they should be aware of the throttling behavior —but I doubt they want to do that.

> Does the stacking feature respect our disable-motion config? (which is activated in tests and makes all transitions instant)

Yes, just manually tested to double-check. These delays have nothing to do with animations.

Reply:
Thanks for linking that PR, because that one is problematic. 

For this test my recommendation would be to wrap the last line into `waitForAssertion`. No any arbitrary delays in between (because real users wouldn't wait the same)

Reply:
They _would_ wait the same, because it's the exact delay that is applied to the focus change.

This is not the same situation as when waiting for the page to e.g, react to user input or submitting a form, where you don't know how much it will take and you want to continue as soon as possible, with a timeout to fail when the duration would be considered too long. I understand that for these typical cases something like `useForAssertion` is preferable.

Reply:
Even if the delay is deterministic, by definition of e2e test it should not rely in implementation detail (so the test passes before and after any non-breaking-change refactorings)

Reply:
But a non-breaking refactoring should not change that delay. And the other way around, the way I think of this, changing the delay should be considered a breaking change, so if the focus change takes shorter or longer unintentionally, that should be caught.

Reply:
Not a big deal. You may keep it this way if you want. But remove the custom `wait`, please

Reply:
Thanks. Updated to use `page.pause` and removed the custom `wait` function.